### PR TITLE
 Fixed simple code warnings reported by Sonar

### PR DESCRIPTION
**Pull request by snakefoot**

### CODE REVIEW

File: src/NLog/Internal/StreamHelpers.cs [hunk 1]
```diff
@@ -96,7 +95,7 @@ public static void CopyWithOffset(this Stream input, Stream output, int offset)
         {
             if (offset < 0)
             {
-                throw new ArgumentException("negative offset");
+                throw new ArgumentException("cannot be negative", nameof(offset));
```
Comment (304NotModified) [inline src/NLog/Internal/StreamHelpers.cs:98]: Maybe this should be a ArgumentOutOfRangeException?

File: src/NLog/LogLevel.cs [hunk 2]
```diff
@@ -251,7 +251,7 @@ public static LogLevel FromOrdinal(int ordinal)
                     return Off;
 
                 default:
-                    throw new ArgumentException("Invalid ordinal.");
+                    throw new ArgumentException($"Unknown loglevel: {ordinal.ToString()}.", nameof(ordinal));
```
Comment (304NotModified) [inline src/NLog/LogLevel.cs:254]:
```suggestion
                    throw new ArgumentException($"Unknown log level ordinal: {ordinal.ToString()}.", nameof(ordinal));
```

**Review by 304NotModified — APPROVED** [review]

**Comment by 304NotModified** [comment]
2 suggestions :)

File: src/NLog/Internal/StreamHelpers.cs (hunk 1 above)
Comment (snakefoot) [reply src/NLog/Internal/StreamHelpers.cs:98]: Just working on removing false positives. So real errors are not hidding in the noise.

File: src/NLog/LogLevel.cs (hunk 2 above)
Comment (snakefoot) [reply src/NLog/LogLevel.cs:254]: Well the Sonar Rule checks if the ArgumentException message contains the input-parameter in the text. So your suggestions wants this instead: `throw new ArgumentException($"Unknown log level {nameof(ordinal}: {ordinal.ToString()}.", nameof(ordinal));`

But I'm fine with just using ArgumentException-parameterName

**Pull request merged**